marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-card.test.tsx

```diff
@@ -126,4 +127,74 @@ describe("Entity Tiles component", () => {
       expect(getByText('Batch Size:')).toBeInTheDocument();
       expect(getByPlaceholderText('Please enter batch size')).toHaveValue('50');
   });
-});
+
+  test('Verify Mapping card allows step to be added to flow with writeStep authority', async () => {
+    const authorityService = new AuthoritiesService();
+    authorityService.setAuthorities(['readMapping','writeStep']);
+    let entityModel = data.primaryEntityTypes.data[0];
+    let mapping = data.mappings.data[0].artifacts;
+    const mockAddStepToFlow = jest.fn();
+    const noopFun = jest.fn();
+    const {getByText} = render(<AuthoritiesContext.Provider value={authorityService}><MappingCard
+      data={mapping}
+      flows={data.flows.data}
+      entityTypeTitle={entityModel.entityName}
+      getMappingArtifactByMapName={noopFun}
+      deleteMappingArtifact={noopFun}
+      createMappingArtifact={noopFun}
+      updateMappingArtifact={noopFun}
+      canReadOnly={authorityService.canReadMapping()}
+      canReadWrite={authorityService.canWriteMapping()}
+      canWriteFlow={authorityService.canWriteStep()}
+      entityModel={entityModel}
+      addStepToFlow={mockAddStepToFlow}
+      addStepToNew={noopFun}/>
+    </AuthoritiesContext.Provider>);
+
+    fireEvent.mouseOver(getByText(mapping[0].name));
+
+    // test adding to existing flow
+    expect(getByText('Add step to an existing flow')).toBeInTheDocument();
+    fireEvent.click(getByText('Select Flow'));
+    fireEvent.click(getByText(data.flows.data[0].name));
+    fireEvent.click(getByText('Yes'));
+    expect(mockAddStepToFlow).toBeCalledTimes(1);
+
+    // adding to new flow
+    fireEvent.mouseOver(getByText(mapping[0].name));
+    expect(getByText('Add step to a new flow')).toBeInTheDocument();
```

Comment:
Nice. Can we also add a check for "Add step to an existing flow"

Reply:
"Add step to an existing flow" is above. It doesn't appear that "Add step to a new flow" is fully implemented yet.